### PR TITLE
Add crawler for Canoas/RS

### LINKED ISSUE #10: Canoas/RS
- Site do Diário de Canoas: http://sistemas.canoas.rs.gov.br/gt/publico/dof/index.jsf
- Gist do testes que fiz: https://gist.github.com/Lrcezimbra/49a8cb3135e4f043212ab9976b3f56a7

Comecei a fazer o crawler de Canoas/RS, porém encontrei alguns problemas. O site do diário de Canoas não disponibiliza link direto para o PDF, ao invés disso usa cookie e sessão para o acesso e isso complica um pouco as coisas.

Como funciona: 
Quando é acessado o site e escolhido o diário que deseja acessar, são feitas algumas requisições para o servidor com o dia e as páginas escolhidos. E então retorna o arquivo PDF. Caso acesse diretamente o link do arquivo, o server verifica a sessão e retorna o PDF escolhido nas requisições anteriores ou 404. Não existe um caminho único para o arquivo, o [link](http://sistemas.canoas.rs.gov.br/gt/javax.faces.resource/dynamiccontent.properties.jsf?ln=primefaces&v=5.3&v=5.3&pfdrid=MLSBZwVJsqaR6Pg%2Bmbf71sxZiO68niDGWIogDS8U9wo%3D&pfdrt=sc&pfdrid_c=true) é o mesmo para todos os arquivos, o que muda é só a data e páginas _setados_ na sessão. 

Então é necessário manter uma sessão única para cada dia do diário, pelo que eu vi é possivel fazer isso com o Scrapy. Porém, no projeto atualmente é rodado o crawler que pega todos os diários e as URLs dos arquivos e depois passa para a próxima pipeline que baixa os arquivos. Porém acredito que nessa troca de pipeline não se mantenha os cookies e a sessão e como nesse caso seria necessário manter a sessão para fazer o download, creio que não funcionaria. Qual vcs acham que é a melhor forma de resolver isso?

Outro problema é que não teremos o link direto para o arquivo do diário como temos de Porto Alegre.

### PR DESCRIPTION
**Pull request by vitorbaptista**
I merged the code from #91 by @Lrcezimbra into main and fixed the case of parsing a day without gazette.

This is working and ready to be merged.

Fixes #10

### CODE REVIEW

File: processing/data_collection/gazette/spiders/rs_canoas.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RsCanoasSpider(BaseGazetteSpider):
+    TERRITORY_ID = "4304606"
+    name = "rs_canoas"
+    START_DATE = date(2018, 5, 29)
```
Comment (jvanz) [inline processing/data_collection/gazette/spiders/rs_canoas.py:13]: Can you use the `start_date` from the base class?

**Review by jvanz — CHANGES_REQUESTED** [review]

File: processing/data_collection/gazette/spiders/rs_canoas.py (hunk 1 above)
Comment (vitorbaptista) [reply processing/data_collection/gazette/spiders/rs_canoas.py]: Done

**Comment by jvanz** [comment]
@vitorbaptista sorry for the delay. Could you rebase the branch in the latest main? Thus, I can merge the PR. Thanks!

**Review by jvanz — APPROVED** [review]

**Comment by vitorbaptista** [comment]
@jvanz Rebased :+1:

**Pull request merged**